client/src/components/PlayerViews/Waiting.vue

```diff
@@ -1,3 +1,172 @@
 <template>
-  <div>Waiting for game to begin...</div>
-</template>
\ No newline at end of file
+  <div class="center">
+    <header class="center" :style="headerColor">
+      <h1>{{ name }}</h1>
+      <v-img
+        :src="require(`../../../assets/pfps/${selectedPfp}.webp`)"
+        class="selected-pfp"
+      ></v-img>
+    </header>
+    <h2 class="my-2">Choose Your Color</h2>
+    <div style="width: 85%">
+      <div class="flex-container">
+        <div v-for="color in colors" :key="color">
+          <div style="position: relative;" >
+            <v-icon 
+              v-if="color === selectedColor"
+              class="check-mark"
+              dark
+            >mdi-check-outline</v-icon>
+            <div
+              @click.stop="selectedColor = color"
+              :style="`background-color: ${color};`"
+              class="frame-item ma-3"
+            ></div>
+          </div>
+        </div>
+      </div>
+    </div>
+    <h2 class="my-2">Take Your Pic</h2>
+    <div style="width: 85%">
+      <div class="flex-container">
+        <div 
+          v-for="i in numOfPfps" 
+          :key="i"
+        >
+          <v-img
+            @click.stop="selectedPfp = i"
+            :src="require(`../../../assets/pfps/${i}.webp`)"
+            :style="pfpSelected(i)"
+            class="frame-item ma-3"
+          ></v-img>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    socketInstance: {
+      required: true,
+      type: Object
+    },
+    clientId: {
+      required: true,
+      type: Number
+    }
+  },
+  mounted() {
+    // selects random starting color
+    const RAND_INX = Math.floor(Math.random() * this.colors.length)
+    this.selectedColor = this.colors[RAND_INX]
+
+    // selects random starting pfp
+    this.selectedPfp = Math.floor(Math.random() * this.numOfPfps) + 1
```

Comment:
Comment should explain why there's a +1  as well to this random math.

Reply:
the pfps are labelled 1-8 and if there was not a plus 1 it would spit out a 0-7 range. Will clear that up